### PR TITLE
Added integration tests for login edge cases.

### PR DESCRIPTION
**Pull request by katybaulch**
Added integration tests for user login.

### CODE REVIEW

File: integration_tests/setup_db.py [hunk 1]
```diff
@@ -253,10 +255,14 @@ def _setup_organisation(test_db: Session) -> int:
         "Test",
         org.id,
         "$2b$12$XXMr7xoEY2fzNiMR3hq.PeJBUUchJyiTfJP.Rt2eq9hsPzt9SXzFC",
-        is_active=True,
     )
+    _add_app_user(test_db, "test1@cpr.org", "TestInactive", org.id, is_active=False)
```
Comment (diversemix) [inline integration_tests/setup_db.py:259]: Doesn't this user also have an empty hashed password?

File: integration_tests/setup_db.py [hunk 2]
```diff
@@ -253,10 +255,14 @@ def _setup_organisation(test_db: Session) -> int:
...
+    _add_app_user(
+        test_db, "test2@cpr.org", "TestHashedPassEmpty", org.id, hashed_pass=""
+    )
+    _add_app_user(test_db, "test3@cpr.org", "TestPassMismatch", org.id, hashed_pass="")
```
Comment (diversemix) [inline integration_tests/setup_db.py:263]: And this one also has an empty hash?

File: integration_tests/setup_db.py (hunk 1 above)
Comment (katybaulch) [reply integration_tests/setup_db.py:260]: Technically, we test whether the user is active or not before we check the validity of their password against the hash.

**Review by diversemix — APPROVED** [review]
Sweet :candy:
